giskard_vision/detectors/example_manager.py

```diff
@@ -0,0 +1,73 @@
+import base64
+from abc import ABC, abstractmethod
+from copy import deepcopy
+from typing import Any
+
+import cv2
+
+
+class ExampleManager(ABC):
+    """
+    Abstract class to manage examples from different data types.
+    """
+
+    @abstractmethod
+    def add_examples(self, example: Any): ...
+
+    @abstractmethod
+    def head(self, n: int): ...
+
+    @abstractmethod
+    def to_html(self): ...
+
+
+class ImagesExampleManager:
+    """
+    Class to manage images examples
+    """
+
+    def __init__(self, examples: list = [], embed: bool = True):
+        self.examples = examples
```

Comment:
```suggestion
    def __init__(self, examples: Optional[list] = None, embed: bool = True):
        self.examples = examples if examples else []
```

Reply:
could you please also rename `*Manager` to `ScanExamples` to be conform with https://github.com/Giskard-AI/giskard/pull/1763/files#diff-f55f5643bb9603a6b308af22a98e376fb581c974aff0f7d8354fe77c2308b528